src/reducers/courses.js

```diff
@@ -79,6 +79,7 @@ function addQueueToCourse(state, courseId, queue) {
 
 function removeQueueFromCourse(state, courseId, queueId) {
   if (
+    !state.courses[courseId] ||
     !state.courses[courseId].queues ||
     !(courseId in state.courses) ||
```

Comment:
I don't think we need this line anymore

Reply:
Good call - looks like I was trying to do the right thing the first time but messed up the short-circuiting. I'll fix the other functions here too.